--- internal/api/handlers.go
+++ internal/api/handlers.go
@@ -337,6 +337,10 @@
 		fail(w, err)
 		return
 	}
+	if !a.Database.HasValidLicense(refereeId) {
+		fail(w, types.ErrForbidden)
+		return
+	}
 
 	var payload PayloadAddAvailability
 	if err := loadPayload(&payload, r.Body); err != nil {
@@ -959,6 +963,10 @@
 		fail(w, dbErr)
 		return
 	}
+	if !a.Database.HasValidLicense(refereeID) {
+		fail(w, types.ErrForbidden)
+		return
+	}
 
 	var payload PayloadRespondToAssignment
 	if err := loadPayload(&payload, r.Body); err != nil {
@@ -1081,6 +1089,10 @@
 		fail(w, dbErr)
 		return
 	}
+	if !a.Database.HasValidLicense(refereeID) {
+		fail(w, types.ErrForbidden)
+		return
+	}
 
 	var payload PayloadSubmitScore
 	if err := loadPayload(&payload, r.Body); err != nil {
